packages/pigeon/lib/dart_generator.dart

```diff
@@ -92,6 +92,24 @@ void _writeCodec(Indent indent, String codecName, Api api) {
   });
 }
 
+/// Creates a Dart type where all type arguments are [Objects].
+String _makeGenericTypeArguments(TypedEntity entity, String nullTag) {
+  return entity.typeArguments != null
+      ? '${entity.dataType}<${entity.typeArguments!.map((TypeDeclaration e) => 'Object$nullTag').reduce(_commaJoin)}>'
```

Comment:
Isn't `.reduce(_commaJoin)` just `.join(', ')`?

Reply:
Yep, almost.  `join` accepts empty lists.  Dart's reduce function doesn't have an initial value parameter like other do so it can't handle empty lists.  However map().reduce() is a construct people are more used to reading than map().join().  I'm even surprised that `join` is a function on `Iterable`.  I would have made it `String String.join(Iterable<String>, String)`  Using `join` on anything other than Iterable<String> is suspect.

Reply:
Yea, I mean if you google "map filter reduce" you'll get hundreds of articles written about using those functions across tons of different languages.  map+reduce even inspired the parallel processing model MapReduce.  You don't have that ubiquity with the join function.

The join function is weird because it hides toString calls inside of its implementation where map+reduce uses the type system to enforce that you actually have Strings.  It's just better imo.